gym/vector/utils/info_strategies.py

```diff
@@ -0,0 +1,99 @@
+"""
+Strategies for processing the info dictionary of VecEnvs
+"""
+from abc import ABC, abstractmethod
+from enum import Enum
+
+from gym.error import InvalidInfoFormat
+
+
+class VecEnvInfoStrategy(ABC):
+    """Interface for implementing a info
+    processing strategy for vectorized environments
+    """
+
+    @abstractmethod
+    def __init__(self, num_envs: int):
+        ...
+
+    @abstractmethod
+    def add_info(self, info: dict, env_num: int):
+        """Get the info dict from the
+        environment and process with the defined strategy
+        """
+
+    @abstractmethod
+    def get_info(self):
+        """Return the info for the
+        vectorized env
+        """
+
+
+class ClassicVecEnvInfoStrategy(VecEnvInfoStrategy):
+    """Process the info dictionary of an environment
+    so that the vectorized info is returned in the form of a list of dictionaries.
+
+    Example with 3 environments:
+        [{}, {}, {}]
+    """
+
+    def __init__(self, num_envs: int):
+        self.num_envs = num_envs
+        self.info = []
+
+    def add_info(self, info: dict, env_num: int):
+        self.info.append(info)
+
+    def get_info(self) -> list:
+        return self.info
+
+
+class BraxVecEnvInfoStrategy(VecEnvInfoStrategy):
+    """Process the info dictionary of an environment
+    so that the vectorized info is returned in the form of a single dictionary.
+    Keys of the dictionary represents the `info` key; Values are lists
+    in which each index correspond to an environment. If the environment
+    at index `i` does not have a value for `info` then it is set to `None`
+
+    This strategy matches Brax library info's output structure.
+
+    Example with 3 environments in which only the last has the `terminal_observation` info:
+        {
+            "terminal_observation": [
+                None,
+                None,
+                array([0.13,  1.58 , -0.22, -2.56])
+            ]
+        }
+    """
+
+    def __init__(self, num_envs: int):
+        self.num_envs = num_envs
+        self.info = {}
+
+    def add_info(self, info: dict, env_num: int):
+        for k in info.keys():
+            info_array = self.info.get(k, [None for _ in range(self.num_envs)])
+            info_array[env_num] = info[k]
+            self.info[k] = info_array
+
+    def get_info(self) -> dict:
+        return self.info
+
+
+class StrategiesEnum(Enum):
+    classic: str = "classic"
+    brax: str = "brax"
+
+
+def get_info_strategy(info_format: str) -> VecEnvInfoStrategy:
+    strategies = {
```

Comment:
Should these be global variables as if people wants their own custom Info Strategy, I dont think it is possible currently.

Reply:
Good catch! I didn't think about it. :+1:  I can switch back to the plain old dictionary as it was before and do the validation inside the env or, maybe something like that? 
```python
class InfoStrategyFactory():
    strategies = {
        InfoStrategiesEnum.classic.value: ClassicVecEnvInfoStrategy,
        InfoStrategiesEnum.brax.value: BraxVecEnvInfoStrategy,
    }

    def get_info_strategy(info_format: str) -> VecEnvInfoStrategy:
        if info_format not in InfoStrategyFactory.strategies:
            raise InvalidInfoFormat(
                "%s is not an available format for info, please choose one between %s"
                % (info_format, list(InfoStrategyFactory.strategies.keys()))
            )
        return InfoStrategyFactory.strategies[info_format]
```

With this approach the user can add a new strategy with:
```python
from gym.vector.utils import InfoStrategyFactory
InfoStrategyFactory.strategies['my_strategy'] = MyNewStrategy
envs = gym.vector.make("CartPole-v1", asynchronous=False, num_envs=3, info_format="my_strategy")
```

Then we can also add a setter method in the class with validation to check if the added strategy matches the InfoStrategy interface  :thinking: 

```python
from gym.vector.utils import InfoStrategyFactory
InfoStrategyFactory.add_strategy('my_strategy', MyNewStrategy)
envs = gym.vector.make("CartPole-v1", asynchronous=False, num_envs=3, info_format="my_strategy")
```

Reply:
Cool, Im good with that solution, we just need to explain it in the docs

Reply:
Could you add a test for `add_strategy`